internal/resource/url.go

```diff
@@ -352,9 +352,15 @@ func (f *Fetcher) FetchFromS3(u url.URL, dest *os.File, opts FetchOptions) error
 
 	sess.Config.Region = aws.String(region)
 
+	var versionId *string
+	if v, ok := u.Query()["versionId"]; ok && len(v) > 0 {
+		versionId = aws.String(v[0])
```

Comment:
Edge case: what happens if the resulting string is `""`? Should we skip setting `versionId` in that case?

Reply:
Nice catch, I believe S3 will ignore an empty version, but to avoid any issues I've added a check for this case and will set `versionId = nil` and log an error in case the versionId content is empty.

Reply:
FFR, S3 returns an error when `?versionId=` is empty:

```
λ aws s3api get-object --bucket a-bucket --key some/key --version-id "" /dev/stdout                                          

An error occurred (InvalidArgument) when calling the GetObject operation: Version id cannot be the empty string

```

Reply:
I will leave this for a second feedback from @ajeddeloh. I think we could follow a similar path of hard-erroring here, and also adding the same check in `validateURL`.